src/utils/api-requests.ts

```diff
@@ -0,0 +1,93 @@
+import { githubArgs } from './env';
+import { Interval, DateTime } from 'luxon';
+import { Octokit } from '@octokit/rest';
+import { WorkflowRun } from '../models/workflow-run';
+import debugBase from '../utils/debug';
+
+const MAX_PER_PAGE = 100;
+
+type WorkflowRunResults = {
+  runs: WorkflowRun[];
+  meta: {
+    per_page: number;
+    total_pages: number;
+  };
+};
+
+export async function fetchWorkflowRuns(
+  interval: Interval,
+  workflowId: number | string,
+  per_page = MAX_PER_PAGE
+): Promise<WorkflowRunResults> {
+  const debug = debugBase.extend('api:fetch-workflow-runs:' + workflowId);
+  let { repo, owner, token } = githubArgs();
+  let client = new Octokit({ auth: token });
+
+  let runData = [];
+  const status = 'success';
+
+  let didLogCount = false;
+  let page = 0;
+
+  for await (const response of client.paginate.iterator(
+    client.rest.actions.listWorkflowRuns,
+    {
+      repo,
+      owner,
+      workflow_id: workflowId,
+      status,
+      per_page,
+    }
+  )) {
+    if (!didLogCount) {
+      didLogCount = true;
+      debug(`found ${response.data.total_count} results`);
+    }
+    page++;
+
+    let first = DateTime.fromISO(response.data[0].created_at).toUTC();
+    let last = DateTime.fromISO(
+      response.data[response.data.length - 1].created_at
+    ).toUTC();
+
+    debug(
+      `page #${page} ${response.data.length} items from ${first} -> ${last}`
+    );
+
+    let exitLoop = false;
+    for (let workflowRunData of response.data) {
+      let createdAt = DateTime.fromISO(workflowRunData.created_at);
+      if (interval.isBefore(createdAt)) {
```

Comment:
why are we checking for this?

Reply:
we discussed and changed this to a labeled outer loop + an inner switch